File: Professional Image Modifier 5/[VB]-Professional Image Modifier 5/VB/ImageFunctions/Controls/MoravecCornerProperties.vb
Imports System.Collections.Generic
Imports System.Windows.Forms

Imports AForge
Imports AForge.Imaging
Imports ImageFunctions.Forms

Namespace ImageFunctions.Controls
	Partial Public Class MoravecCornerProperties
		Inherits UserControl

		Public Property WindowDefault() As Integer
		Public Property ThresholdDefault() As Integer
		Public Property Window() As Integer
		Public Property Threshold() As Integer

		Public Sub New()
			InitializeComponent()
			ThresholdDefault = 500
			WindowDefault = 3
			trackThreshold.Value = ThresholdDefault
			trackWindow.Value = WindowDefault
		End Sub

		Public Sub SetDefaults()
			trackThreshold.Value = ThresholdDefault
			trackWindow.Value = WindowDefault
		End Sub

		Private Sub trackThreshold_ValueChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles trackThreshold.ValueChanged
			lblThreshold.Text = trackThreshold.Value.ToString()
			Threshold = trackThreshold.Value
		End Sub

		Private Sub trackWindow_ValueChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles trackWindow.ValueChanged
			lblWindow.Text = trackWindow.Value.ToString()
			Window = trackWindow.Value
		End Sub
	End Class
End Namespace

File: Professional Image Modifier 4/[C#]-Professional Image Modifier 4/C#/ImageFunctions/Controls/MoravecCornerProperties.cs
using System.Collections.Generic;
using System.Windows.Forms;

using AForge;
using AForge.Imaging;
using ImageFunctions.Forms;

namespace ImageFunctions.Controls
{
	public partial class MoravecCornerProperties : UserControl
	{

		public int WindowDefault { get; set; } // As set by Aforge
		public int ThresholdDefault { get; set; }  // As set by Aforge
		public int Window { get; set; } // Set by User
		public int Threshold { get; set; } // Set by User

		public MoravecCornerProperties()
		{
			InitializeComponent();
			ThresholdDefault = 500;
			WindowDefault = 3;
			trackThreshold.Value = ThresholdDefault;
			trackWindow.Value = WindowDefault;
		}

		public void SetDefaults()
		{
			trackThreshold.Value = ThresholdDefault;
			trackWindow.Value = WindowDefault;
		}

		private void trackThreshold_ValueChanged(object sender, System.EventArgs e)
		{
			lblThreshold.Text = trackThreshold.Value.ToString();
			Threshold = trackThreshold.Value;
		}

		private void trackWindow_ValueChanged(object sender, System.EventArgs e)
		{
			lblWindow.Text = trackWindow.Value.ToString();
			Window = trackWindow.Value;
		}
	}
}
